Lib/test/test_asyncore.py

```diff
@@ -800,6 +801,44 @@ def test_quick_connect(self):
             if t.is_alive():
                 self.fail("join() timed out")
 
+    def test_map_altered_in_loop(self):
+        family = self.family
+        fail = self.fail
+
+        class PoisonedClient(BaseClient):
+            """
+            This dispatcher is created after closing a writable one
+            """
+            def handle_write(self):
+                fail("Attempt to call handle_write on the wrong client")
+
+            # The dispatcher is not writable and therefore handle_write shouldn't be called
+            def writable(self):
+                return False
+
+        class ManagerClient(BaseClient):
+
+            def __init__(self, family, address):
+                BaseClient.__init__(self, family, address)
+                self.old_client = BaseClient(family, address)
+
+            def handle_write(self):
+                old_fd = self.old_client._fileno
+                self.old_client.close()
+                # This trusts that the fd of this client new
+                # will be the same to the one just closed
+                new_client = PoisonedClient(family, server.address)
+                if new_client._fileno != old_fd:
+                    raise unittest.SkipTest("The test is meaningful only if the fd for the old and "
+                                            "the new dispatcher are the same")
+
+                self.flag = True
```

Comment:
I don't think I understand this, if the first readable object closes the second and removes it from the map then it will hit the comparison
`if obj is None:`
 and the second readable object won't be called as it shouldn't.